Newsletters: add 139 (2021-03-10)

harding opened the pull request:
- [ ] Add releases/RCs @harding 
- [ ] Add topic links when all content done @harding
- [ ] Bitcoin Core PR Review Club summary @jonatack or @jnewbery 
- [ ] `Bitcoin Core #20685` Add I2P support using I2P SAM @adamjonas 
- [ ] `BOLTs #839` Add 2016 blocks channel funding timeout @dongcarl
- [ ] `BTCPay Server #2181` Reenabling uppercase BECH32 in QR codes @Xekyo

Xekyo commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -158,7 +158,11 @@ BOLTs][bolts repo].*
-- [BTCPay Server #2181][] Reenabling uppercase BECH32 in QR codes FIXME:Xekyo
+- [BTCPay Server #2181][] uppercases bech32 addresses when presenting Bitcoin
+  URIs as QR codes. This results in [less dense QR codes][bech32 uppercase qr]
+  as strings starting or ending with uppercase substrings can be encoded more
```
Mh, given that they still use `bitcoin:` as the prefix, maybe the claim should be attenuated a bit.

harding replied:
QR codes have a "mixed" mode where they can shift from one encoding to another multiple times at a cost per shift of about 20 bits, so the more contiguous uppercase characters you can add, the better, no matter where they occur in the string.  See "However, QR codes can support mixed character sets [...]" in https://bitcoinops.org/en/bech32-sending-support/#creating-more-efficient-qr-codes-with-bech32-addresses

Xekyo replied:
Oh, cool, I understood that as only working for sequences at the start or end.

Xekyo commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,186 @@
+## News
+
+- **Taproot activation discussion:** the previous weeks' discussions
+  about activation saw different groups of people opposing using either
```
"opposing using" makes me pause

```suggestion
  about activation saw different groups of people opposing either
```

Xekyo reviewed (commented):
Looks good to me so far

adamjonas commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -145,7 +145,13 @@ repo], [Hardware Wallet Interface (HWI)][hwi repo],
+  [long been requested][Bitcoin Core #2091] and was only recently made
+  possible by the addition of [addr v2][addr v2 newsletter mention].
+  Though documentation for node operators hoping to run I2P is still
+  being created, a few [IRC discussions][I2P install comment] provide
```
I'm not too confident about adding these IRC discussions but thought it might be nice in the absence of documentation.

jonatack replied:
(if useful, a how-to in one tweet: https://twitter.com/jonatack/status/1366764964896075776, just needs to be i2pd version 2.35 and up)

jonatack replied:
or here https://github.com/bitcoin/bitcoin/pull/20685#pullrequestreview-601031442 though it's not really explicit

michaelfolkson replied:
@jonatack: Cool, added a SE question: https://bitcoin.stackexchange.com/questions/103402/how-can-i-use-bitcoin-core-with-the-anonymous-network-protocol-i2p

jonatack replied:
Thanks! Very handy. One error and a couple changes: `onlynet=tor` should in fact be `onlynet=onion`, and maybe drop the `debug=tor` and `onlynet=onion` (I was replying on twitter to someone asking how to run both together, could keep the tweet link though). Finally, the version of the `i2pd` router apparently needs to be 2.35 and up (2.36 was recently released), at least for now--further I2P code changes are in the works.

jonatack replied:
There are 2 config options for I2P:
```
  -i2pacceptincoming
       If set and -i2psam is also set then incoming I2P connections are
       accepted via the SAM proxy. If this is not set but -i2psam is set
       then only outgoing connections will be made to the I2P network.
       Ignored if -i2psam is not set. Listening for incoming I2P
       connections is done through the SAM proxy, not by binding to a
       local address and port (default: 1)

  -i2psam=<ip:port>
       I2P SAM proxy to reach I2P peers and accept I2P connections (default:
       none)
```

jonatack replied:
I guess I really need to write a doc...

michaelfolkson replied:
@jonatack: Updated, thanks. If you have any further edits, corrections feel free to comment on the SE answer.

jonatack replied:
@michaelfolkson thanks--much appreciated. I don't have an SE account (like LinkedIn, I've never had one) and don't think I can comment without one.

bitschmidty commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,196 @@
+[news107 decthresh]: /en/newsletters/2020/07/22/#mailing-list-thread
+[news137 bip8conf]: /en/newsletters/2021/02/24/#taproot-activation-discussion
+[btcpay uri survey]: https://github.com/btcpayserver/btcpayserver/issues/2110
+[bech32 uppercase qr]: https://bitcoinops.org/en/bech32-sending-support/#creating-more-efficient-qr-codes-with-bech32-addresses
```
Remove "https://bitcoinops.org" portion

Xekyo replied:
Okay, will fix

bitschmidty reviewed (commented):
So far so good, one comment

jnewbery commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,202 @@
+- [Bitcoin Core #20685][] Adds I2P support by using the
+  [I2P SAM protocol][I2P SAM protocol].  This feature has
+  [long been requested][Bitcoin Core #2091] and was only recently made
+  possible by the addition of [addr v2][addr v2 newsletter mention].
```
Would it be better to link to the addr v2 topic here, rather than just the PR merge coverage?

jnewbery commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,202 @@
+  funders fees when funding transactions fail to confirm and provide stronger
+  guarantees for the channel funder and fundee. The new recommendations suggest
+  that the funder commits to ensuring the funding transaction confirms in 2016
+  blocks and that the fundee forgets the pending channel should the funding
+  transaction not confirm within 2016 blocks.
```
Perhaps "if {thing} does not" as simpler English than "should {thing} not".

jonatack commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -119,14 +119,61 @@ existing software building on top of taproot.
+     security. The idea was presented in a 2019 paper, *[Bandwidth-Efficient
+     Transaction Relay for Bitcoin][erlay paper]*, and is specified in
+     [BIP330][]."
+  a0link="https://github.com/naumenkogs/bips/blob/bip_0330_updates/bip-0330.mediawiki#erlay"
```
This line should be removed. Will re-push.

jonatack commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -119,14 +119,61 @@ existing software building on top of taproot.
+     connections][erlay 2], making the network more robust to partitioning
+     attacks and [single nodes more resistant to Eclipse attacks][erlay 3];
+     finally, improved privacy by no longer flooding transactions from private
+     nodes."
```
@naumenko I'm not 100% sure about this line about privacy, accuracy check please if you have some spare (txn) bandwidth

jnewbery replied:
I think I'd leave this out. I think Erlay may help privacy marginally, but it's hard to quantify and is not the primary motivation.

jonatack replied:
Thanks, that's better--done.

jnewbery commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,254 @@
+
+  q5="How would 2 peers reach agreement on using Erlay's set reconciliation?"
+  a5="Via a new `sendrecon` peer-to-peer message exchanged after receiving a
+     `wtxrelay` message."
```
Perhaps s/after receiving a wtxrelay message/during the version-verack handshake/

I think the fact that it's after wtxidrelay is more of a low-level detail. The important thing is that it's during the version-verack handshake.

If you do leave this as it is, there's a typo: wtxrelay -> wtxidrelay.

jonatack replied:
Agree; done.

jnewbery reviewed (commented):
Review Club summary looks good. Thanks @jonatack. Just a couple of minor comments.

harding commented:
Reviewed everyone's contributions, thanks!  Only a few trivial edits made.  Also made edits feedback, added releases/RCs, and added topic links.  Thanks everyone!

jonatack reviewed (commented):
Just started reading through (not checking links, bad connection) and saw this was re-pushed, so putting this up so far. Edit: maybe was not just re-pushed, but the "refresh" notice just arrived :shrug:

jonatack commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,281 @@
+[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
+BOLTs][bolts repo].*
+
+- [Bitcoin Core #20685][] Adds I2P support by using the
```
```suggestion
- [Bitcoin Core #20685][] adds I2P support by using the
```
maybe "add support for the I2P privacy network using the..."

jonatack commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,281 @@
+- [Bitcoin Core #20685][] Adds I2P support by using the
+  [I2P SAM protocol][I2P SAM protocol].  This feature has
+  [long been requested][Bitcoin Core #2091] and was only recently made
+  possible by the addition of [addr v2][topic addr v2]
```
```suggestion
  possible by the addition of [addr v2][topic addr v2].
```

jonatack commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,281 @@
+- [BOLTs #839][] adds funding transaction timeout recommendations to save
+  funding fees when there's a failure to confirm funding transactions, providing stronger
+  guarantees for the channel funder and fundee. The new recommendations suggest
+  that the funder commits to ensuring the funding transaction confirms in 2016
```
```suggestion
  that the funder commit to ensuring the funding transaction confirms in 2016
```

jonatack commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,281 @@
+  funding fees when there's a failure to confirm funding transactions, providing stronger
+  guarantees for the channel funder and fundee. The new recommendations suggest
+  that the funder commits to ensuring the funding transaction confirms in 2016
+  blocks and recommends that the fundee forget the pending channel if the funding
```
```suggestion
  blocks and recommend that the fundee forget the pending channel if the funding
```

jonatack commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,281 @@
+  guarantees for the channel funder and fundee. The new recommendations suggest
+  that the funder commits to ensuring the funding transaction confirms in 2016
+  blocks and recommends that the fundee forget the pending channel if the funding
+  transaction not confirm within those 2016 blocks.
```
```suggestion
  transaction does not confirm within those 2016 blocks.
```

jnewbery replied:
Came here to say the same thing!

jonatack reviewed (commented):
Excellent newsletter and discussion of taproot activation methods :rocket:  A few minor grammar suggestions.

jonatack commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,281 @@
+
+- [Bitcoin Core #20685][] Adds I2P support by using the
+  [I2P SAM protocol][I2P SAM protocol].  This feature has
+  [long been requested][Bitcoin Core #2091] and was only recently made
```
https://github.com/bitcoin/bitcoin/issues/2091 -- good find!

bitschmidty commented:
Nice newsletter team!

⌨️ @harding @Xekyo @adamjonas @dongcarl @jonatack 
🕵️ @jnewbery @michaelfolkson

jnewbery commented:
Great newsletter. Thanks @harding and team!

@bitschmidty - there are a few outstanding review comments that we could perhaps fix up in a follow-up PR?

jonatack commented:
> there are a few outstanding review comments that we could perhaps fix up in a follow-up PR?

I'll open a patch.

harding commented on _posts/en/newsletters/2021-03-10-newsletter.md:
```diff
@@ -0,0 +1,281 @@
+      difficult in the future, both because we'll only do so when we're
+      already in crisis, and because Bitcoin's growth means future
+      agreement will need to be done at greater scale and so with
+      greater difficulty."  <!-- statement written by me trying to
+      summarize Rusty Russell's concerns, then revised by him -->
```
If we have to make a change, I'd rather we drop the quotation marks than attribute Rusty as (1) we don't attribute any of the other criticisms to specific people; (2) this was my summary of the situation that Rusty just edited (which also justifies our using it without quotes if necessary), I'm not sure Rusty would phrase it that way on his own.

If we want, we can link to the log where it was discussed, http://gnusha.org/taproot-activation/2021-03-08.log , unfortunately that log site doesn't allow linking to specific lines, which is why I tend to avoid using it when there's been a busy day or an alternative is available.
